### DISCUSSION TITLE
SERIAL_9N1 was not declared in this scope

### DISCUSSION THREAD

**Discussion by Fietsenwiel** [AK-Homberger/Seatalk-Autopilot-Remote-Control#10] [Q&A] [answered]
I try to compilate the sketch and I copied the HardwareSerial9bit" or "NewHWSworking" to the library directory but still I get the error:
"SERIAL_9N1 was not declared in this scope".

What is going wrong?

Thanks!

**Comment by AK-Homberger** [comment]
Hi Fietsenwiel,
as written in the Software section in Readme.md, the 9 bit implementation is depending on the version (IDE and Bord information).

Did you try this with versions shown in documentation?

Regards,
Andreas

**Comment by Fietsenwiel** [comment]
I tried both.
But now I just replaced SERIAL_9N1 for 0x86 and then it compiles. Is that a valid solution?

**Comment by AK-Homberger** [comment]
No, that's not a solition. The 9 bit protocol is necessary to communicate to Seatalk. It has to compile with SERIAL_9N1. Please change IDE and Board setup until it compiles without error.

**Comment by Fietsenwiel** [comment]
Thanks for your support so far.

But I still don't understand.

I compiled first with the board Leonardo , first with "HardwareSerial9bit" and then with "NewHWSworking" in the library directory.
Then I found and installed the sparkfun pro micro board and I did the same.
But the error stays.

Board info says:
BN: Arduino Leonardo
VID: 2341
PID: 8036

**Comment by Fietsenwiel** [comment]
Maybe I have to include something?

**Reply by AK-Homberger** [reply]
For the Leonardo board please use the Arduino AVR Boards version 1.8.3 with "HardwareSerial9bit". Since version 1.8.4, the code has changed again and the 9 bit implementation is not working.

The documentation is updated accordingly.

**Comment by Fietsenwiel** [comment]
I now put the files of the "HardwareSerial9bit" directory with the ino file.
Then, when compiling I get "rx.bytes is not declared in this scope" in line : _rx.bytes[1] = bit_is_set(*_ucsrb,RXB80)>>RXB80; //get the 9th bit

And when I try the files in "NewHWSworking" than  I get the error: prototype for 'void HardwareSerial:begin(long unsigned int,......  in the line: void hardwareSerial::begin(unsigned long baud, uint16_t config)

I now have no idea to get this working....

**Reply by AK-Homberger** [reply]
The three files from "HardwareSerial9bit" (or "NewHWSworking" for older IDE) have to be copied into the Arduino directory on your local PC e.g.: "C:\Program Files (x86)\Arduino\hardware\arduino\avr\cores\arduino". Not to the directory with the .ini file. This is only tru for Windows sytems!

**Comment by Fietsenwiel** [comment, marked as answer]
Thanks, "For the Leonardo board please use the Arduino AVR Boards version 1.8.3 " did the trick.

Now I will put the components together, can't wait to see the magic.

**Comment by janokylaszlo** [comment]
Hi there, 
I tried to compile the sketch: copied the contents of HardwareSerial9bit next to the ino file, installed the 1.8.3 version of AVR board - but  the process stop with: 
... In function 'void setup()':
... error: 'SERIAL_9N1' was not declared in this scope
.... Compilation error: '_rx' was not declared in this scope
error messages :(

what am I getting wrong, what have I done wrong?
Thanks a lot

**Comment by janokylaszlo** [comment]
Dear Andreas,

thank you very much for your reply!

I copied these 3 files into the
C:\Program Files (x86)\Arduino\hardware\arduino\avr\libraries\HardwareSerial9bit
folder, and the .ino file left alone.
The following error message is displayed during compilation:

Arduino: 1.8.10 (Windows 10), Board: "Arduino Leonardo"
C:\Users\LászlóJánoky\Arduino\SeaTalk_Autopilot\SeaTalk_Autopilot.ino: In function 'void setup()':
SeaTalk_Autopilot:153:24: error: 'SERIAL_9N1' was not declared in this scope
C:\Users\LászlóJánoky\Arduino\SeaTalk_Autopilot\SeaTalk_Autopilot.ino:153:24: note: suggested alternative: 'SERIAL_8N1'

Arduino IDE 1.8.10
Board: Arduino AVR Boards Leonardo 1.8.3

I don`t know what else I van do wrong… :(

best regards
Laszlo JANOKY

On 2023. Mar 1., at 14:40, Andreas Koritnik ***@***.***> wrote:



The three files from "HardwareSerial9bit" (or "NewHWSworking" for older IDE) have to be copied into the Arduino directory on your local PC e.g.: "C:\Program Files (x86)\Arduino\hardware\arduino\avr\cores\arduino". Not to the directory with the .ini file. This is only tru for Windows sytems!

**Comment by janokylaszlo** [comment]
Dear Andreas,

I figured it out: replace the original (default) HardwareSerial* files with the ones in the repo :)

regards
LJ


On 2023. Mar 1., at 14:40, Andreas Koritnik ***@***.***> wrote:



The three files from "HardwareSerial9bit" (or "NewHWSworking" for older IDE) have to be copied into the Arduino directory on your local PC e.g.: "C:\Program Files (x86)\Arduino\hardware\arduino\avr\cores\arduino". Not to the directory with the .ini file. This is only tru for Windows sytems!

**Comment by AK-Homberger** [comment]
I can't re-produce your problems. My only advise now is to try it with a clean install of the IDE: https://forum.arduino.cc/t/uninstall-for-a-fresh-re-install/349503

After a clean install, copy the tree files again to the folder.
